### ISSUE TITLE
Use astropy units

### ISSUE THREAD

**Issue by Didou09** [ToFuProject/datastock#57]
Preliminary:
---------------

- [x] Verify we can go from astropy `units` to `str`
- [x] Verify we can go from `str` to astropy `units`
- [x] Show typical usage examples

Main usage examples:
------------------------------

@adriendaros  could you show here some typical usage examples ?

**Comment by adriendaros** [comment]
Astropy is a well-known and very used library for Astronomy in Python.
Despite this different field from ours, it has very useful subclasses for all, including astropy.constants and astropy.units.
The latter allows you to define units with the desired values not as strings but as quantities in their own right.
The following url link refers to the official documentation provided by Astropy : [https://docs.astropy.org/en/stable/units/index.html](url)

Examples:
-----------

- Assign units to physical variables, multiply themselves and unit convertion:
![image](https://user-images.githubusercontent.com/81628304/187859177-1d02dc27-a5f7-406b-b6eb-83a0e996a8d8.png)

- Assign a specific unit to a value in a dictionnary and work with them, interms of conversion:
![image](https://user-images.githubusercontent.com/81628304/187860505-c7c33568-984e-4a20-a84d-9b01d4c66c7e.png)

**Comment by Didou09** [comment]
This looks great !

Did you have a chance to look at xarray also for comparison ?

**Comment by adriendaros** [comment]
The main goal here is to integrate a good format for units in `Datastock`, in order to be used in the same way as `float`, `int`, etc and to avoid the use of `str` (informative rather than practical).

Another possibility to integrate a appropriate format for units is the library `xarray `:
**Xarray documentation**
[https://docs.xarray.dev/en/stable/index.html](url)
This library provides a different way to build and deal with N-dimensionnal arrays.
According to the units attaching and saving points, the following link details the steps to follow, as well as to build in the appropriate way the arrays:
**Creating and saving objects with units**
[https://pint-xarray.readthedocs.io/en/stable/creation.html](url)
**Converting units**
[https://pint-xarray.readthedocs.io/en/stable/conversion.html](url)

It looks like the units attached to the data variables are integrated as `str` (a point we want to avoid) and no equaly manner to use them as `astropy.units` do is notified so far.

In conclusion, except the fact that the `xarray` library seems to be a powerfull tool to create and manage N-dim arrays, the `astropy.units` package offers a simplier and faster way, for our use of units, to deal with them.
